### PR TITLE
Trashbag Alt Verb to dump into disposals + Cigars are tagged as trash

### PR DESCRIPTION
**Pull request by ProfanedBane**
## About the PR 
Adds an alt verb to dump trash bags/dumpable stuff into a disposals bin, with a higher priority than flushing.
Additionally marks Cigars as trash able to be picked up, matching cigarettes.

**Screenshots**

https://user-images.githubusercontent.com/36012938/175830107-1ee4448b-af86-456d-a92a-053d5e1e95c1.mp4

**Changelog**

:cl:
- add: The contents of trash bags can now be dumped into disposals with alt click, you can then flush by alt clicking again
- tweak: Cigars can now be picked up by trash bags

### CODE REVIEW

File: Content.Server/Disposal/Unit/EntitySystems/DisposalUnitSystem.cs [hunk 1]
```diff
@@ -34,10 +37,12 @@ public sealed class DisposalUnitSystem : SharedDisposalUnitSystem
     {
         [Dependency] private readonly IMapManager _mapManager = default!;
         [Dependency] private readonly IRobustRandom _robustRandom = default!;
+        [Dependency] private readonly IEntityManager _entMan = default!;
```
Comment (EmoGarbage404) [inline Content.Server/Disposal/Unit/EntitySystems/DisposalUnitSystem.cs:40]: Entity Systems already have EntityManager so injecting this dependency isn't needed.

File: Content.Server/Disposal/Unit/EntitySystems/DisposalUnitSystem.cs [hunk 2]
```diff
@@ -75,25 +80,46 @@ public override void Initialize()
...
+                AlternativeVerb ejectVerb = new();
+                ejectVerb.Act = () => TryEjectContents(component);
+                ejectVerb.Category = VerbCategory.Eject;
+                ejectVerb.Text = Loc.GetString("disposal-eject-verb-contents");
+                args.Verbs.Add(ejectVerb);
+            }
+
+            // Behavior if using a trash bag & other dumpable containers
+            if (args.Using != null
+                && EntityManager.HasComponent<DumpableComponent>(args.Using.Value)
```
Comment (EmoGarbage404) [inline Content.Server/Disposal/Unit/EntitySystems/DisposalUnitSystem.cs:109]:
```suggestion
                && HasComp<DumpableComponent>(args.Using.Value)
```

File: Content.Server/Disposal/Unit/EntitySystems/DisposalUnitSystem.cs [hunk 3]
```diff
@@ -75,25 +80,46 @@ public override void Initialize()
...
+                && TryComp<DumpableComponent>(args.Using.Value, out var dumpable)
+                && TryComp<ServerStorageComponent>(args.Using.Value, out var storage)
+                && storage.StoredEntities != null
+                && storage.StoredEntities.Count > 0)
+            {
+                // Verb to dump held container into disposal unit
+                AlternativeVerb dumpVerb = new();
+                dumpVerb.Act = () => _entMan.EntitySysManager.GetEntitySystem<DumpableSystem>().StartDoAfter(args.Using.Value, args.Target, args.User, dumpable, storage);
```
Comment (EmoGarbage404) [inline Content.Server/Disposal/Unit/EntitySystems/DisposalUnitSystem.cs:117]:
```suggestion
                dumpVerb.Act = () => _dumpableSystem.StartDoAfter(args.Using.Value, args.Target, args.User, dumpable, storage);
```
if you have the dependency there's no need to get the system through the entity manager.

**Review by EmoGarbage404 — COMMENTED** [review]
Overall looks good, just a few small style tweaks.

**Comment by ProfanedBane** [comment]
@EmoGarbage404 Thanks for the review, changes have been pushed,

**Review by EmoGarbage404 — APPROVED** [review]

**Pull request merged**